File: frontend/src/types/evidence.ts
import { Severity } from './incident';
import { DetectorResult } from './detector';

export interface EvidencePackage {
  id: string;
  incident_id: string;
  created_at: string;
  incident_summary: string;
  risk_score: number;
  confidence_score: number;
  severity: Severity;
  affected_entity: {
    email: string;
    phone: string;
    ip: string;
    device: string;
  };
  timeline: Array<{
    time: string;
    event: string;
    severity: Severity;
  }>;
  detector_results: Record<string, DetectorResult>;
  recommended_actions: string[];
  explainability_notes: string[];
  analyst_notes?: string;
}

File: backend/schemas/evidence.py
from pydantic import BaseModel
from typing import Optional


class EvidenceResponse(BaseModel):
    id: str
    incident_id: str
    summary: Optional[str] = None
    risk_score: Optional[str] = None
    confidence_score: Optional[str] = None
    affected_entity: Optional[dict] = None
    timeline: Optional[list] = None
    detector_results: Optional[dict] = None
    recommended_actions: Optional[list[str]] = None
    explainability_notes: Optional[list[str]] = None
    created_at: str
